rasterio/_base.pyx

```diff
@@ -449,6 +449,45 @@ cdef class DatasetBase(object):
                     GDALGetRasterUnitType(self.band(j)) for j in self.indexes)
             return self._units
 
+    def block(self, bidx, i, j):
+        """Returns info about a particular block
+
+        Parameters
+        ----------
+        bidx: int
+            Band index, starting with 1.
+        i: int
+            Row index of the block, starting with 0.
+        j: int
+            Column index of the block, starting with 0.
+
+        Returns
+        -------
+        {'window': Window, 'size': int (bytes)}
+        """
+        cdef GDALMajorObjectH obj = NULL
+        cdef char *value = NULL
+        cdef const char *key_c = NULL
+
+        obj = self.band(bidx)
+
+        if self.driver == 'GTiff':
+            key_b = 'BLOCK_SIZE_{0}_{1}'.format(j, i).encode('utf-8')
+            key_c = key_b
+            value = GDALGetMetadataItem(obj, key_c, 'TIFF')
+            size = int(value)
+        else:
```

Comment:
Can you point to a particular image and the exact key you queried ? Can't reproduce on a simple case.